heartbeat/dnsupdate.in

```diff
@@ -59,13 +62,30 @@ This resource agent manages IP take-over via dynamic DNS updates.
 
 <parameter name="hostname" unique="1" required="1">
 <longdesc lang="en">
-The hostname whose IP address will need to be updated.
+Either the hostname whose IP address will need to be updated (in case of type=A)
+or alias whose hostname will need to be updated (in case of type=CNAME).
 </longdesc>
 <shortdesc lang="en">Hostname to update</shortdesc>
 <content type="string" default="${OCF_RESKEY_hostname_default}" />
 </parameter>
 
-<parameter name="ip" unique="0" required="1">
+<parameter name="cname" unique="1" required="0">
+<longdesc lang="en">
+The CNAME whose hostname address will need to be updated.
+</longdesc>
+<shortdesc lang="en">CNAME to update</shortdesc>
+<content type="string" default="${OCF_RESKEY_cname_default}" />
+</parameter>
+
+<parameter name="type" unique="1" required="1">
```

Comment:
Change to `required="0"` after setting default as mentioned above.

Also `unique="1"` will only let you create 1 dnsupdate resource of the type, so you should change that to 0 as well.